### DISCUSSION TITLE
[Share] Remove All Tags of Selection

### DISCUSSION THREAD

**Discussion by Yang1824** [windingwind/zotero-actions-tags#127] [Action Scripts]
Event: None
Operation: Script
Shortcut: ...

```js
// Remove tags
// @author Yang1824
// @link https://github.com/windingwind/zotero-actions-tags/discussions/127
// @usage Shortcut

if (!item) return;
item.getTags().map((tag) => tag.tag).forEach((tag) => {item.removeTag(tag)})
```

**Comment by mjthoraval** [comment]
Thank you for the script.
It is working nicely, but it just shows an error popup:
![image](https://github.com/windingwind/zotero-actions-tags/assets/36054098/f292305d-77b1-4fbc-b024-c0f727a6b61e)
Is it possible to fix this?

Also, is it possible to get it to work also from the Reader menu?
It did not do anything for me from that menu.

**Reply by theSafdarAwan** [reply]
I am getting the same error

**Reply by windingwind** [reply]
Updated

**Reply by mjthoraval** [reply]
Thank you. The error is gone

> Also, is it possible to get it to work also from the Reader menu?
It did not do anything for me from that menu.

I understand now that triggering the action from the Reader menu actually removes all tags from the PDF attachment, not from the parent item.
It is probably a problem from the Zotero UI that it does not show the Tags of the attachment file when looking at the file in the reader.

**Reply by windingwind** [reply]
```js
// Remove tags from parent items
// @author Yang1824, windingwind
// @link https://github.com/windingwind/zotero-actions-tags/discussions/127
// @usage Shortcut

if (!item) return;
let parentItem = Zotero.Items.getTopLevel(item)[0];
parentItem.getTags().map((tag) => tag.tag).forEach((tag) => {parentItem.removeTag(tag)})
```

**Reply by ramonmi** [reply]
> ```js
> // Remove tags from parent items
> // @author Yang1824, windingwind
> // @link https://github.com/windingwind/zotero-actions-tags/discussions/127
> // @usage Shortcut
> 
> if (!item) return;
> let parentItem = Zotero.Items.getTopLevel(item)[0];
> parentItem.getTags().map((tag) => tag.tag).forEach((tag) => {parentItem.removeTag(tag)})
> ```

运行遇到报错items.map is not a function，类似的问题见 https://github.com/windingwind/zotero-actions-tags/discussions/246 ，是否是插件的问题呢？

**Reply by ramonmi** [reply]
> > ```js
> > // Remove tags from parent items
> > // @author Yang1824, windingwind
> > // @link https://github.com/windingwind/zotero-actions-tags/discussions/127
> > // @usage Shortcut
> > 
> > if (!item) return;
> > let parentItem = Zotero.Items.getTopLevel(item)[0];
> > parentItem.getTags().map((tag) => tag.tag).forEach((tag) => {parentItem.removeTag(tag)})
> > ```
> 
> 运行遇到报错items.map is not a function，类似的问题见 #246 ，是否是插件的问题呢？

`let parentItem = Zotero.Items.getTopLevel(item)[0];` 改为 `let parentItem = Zotero.Items.getTopLevel([item])[0];` 即可
